### ISSUE TITLE
setting StaticCacheDuration to time.Duration(1) causes 100% cpu

### ISSUE THREAD

**Issue by boljen** [kataras/iris#479]
- Version : **4.6.0**

- Operating System: ubuntu 16.04

- [x] I have read the [book](https://www.gitbook.com/book/kataras/iris/details), [examples](https://github.com/iris-contrib/examples), [contributing file](https://github.com/kataras/iris/blob/master/.github/CONTRIBUTING.md) and I'm sure that this issue is not posted & answered before.

- [ ] I am an iris [donator](https://github.com/kataras/iris/blob/master/DONATIONS.md).

--------------

Hi, 

First of all thank you for the amazing work you've done with this project.

I might have found a bug;

Acccording to the documentation, setting StaticCacheDuration to time.Duration(1) should disable the cache. Also see https://kataras.gitbooks.io/iris/content/static-files.html of the book:

> Static, StaticFS and StaticWeb functions automatically cache the given files for a period of time (default 20 seconds). In certain situations you don't want that caching to happen (development etc.).

> Caching can be disabled by setting github.com/kataras/iris/config's StaticCacheDuration to time.Duration(1) before calling any of the named functions. Setting StaticCacheDuration to time.Duration(0) will reset the cache time to 10 seconds (as specified in fasthttp).

However, after following the advice, my application suddenly had a continuous 100% cpu load (read: one core exhausted.) I profiled my application and discovered the high cpu usage is caused by the fasthttp library; ![profile_iris](https://cloud.githubusercontent.com/assets/4733114/19380775/f39997b8-91f8-11e6-8005-d838bee07193.png).

The profile results lead me to believe that time.Duration(1) sets the interval at which the cache is cleared, rather than actually disabling caching, causing a continuous iteration, which leads to an exhausted core.

I'm posting this here because I'm not sure whether this is a bug or if the documentation is simply not accurate.

Hope I could be of help.

**Comment by Liamraystanley** [comment]
I'm able to replicate with:
```bash
$ find static/
static/
static/index.html
```

```go
$ cat test.go
package main

import (
	"time"

	"github.com/kataras/iris"
)

func main() {
	iris.StaticCacheDuration = time.Duration(1)
	iris.StaticFS("/static", "./static", 1)
	iris.Listen(":8080")
}
```

and, now just hit one of the static files:
```bash
$ curl -qs http://localhost/static/index.html
```

**Comment by kataras** [comment]
Hmm that's coming from fasthttp, please post it [there](https://github.com/valyala/fasthttp), both of you, I think I am remembering an issue like this but it supposed to be fixed by valyala.

Thanks,
kataras

**Comment by boljen** [comment]
I did some research to submit an issue to the fasthttp team but I can't find where they claim that time.Duration(1) disables caching. Perhaps this was true for older versions, but it doesn't appear to be the case anymore.

This is where fasthttp declares the cache duration;
https://github.com/valyala/fasthttp/blob/master/fs.go#L247

This is where fasthttp parses the cache duration;
https://github.com/valyala/fasthttp/blob/master/fs.go#L331

Am I overlooking something?

> But in any case don't use StaticFS with 1 sec cache dur it's a bad design, I think the 'bug' shows you that you shouldn't use the StaticFS with refresh on every sec..

This was my original snippet, which according to the book should disable caching, but ended up consuming 100% cpu.

	if IsDevelopment {
		iris.Config.IsDevelopment = true
		iris.StaticCacheDuration = time.Duration(1)
	}

Because the default duration was too long for development purposes and 100% cpu isn't acceptable, I rewrote the snippet as follows;

	if IsDevelopment {
		iris.Config.IsDevelopment = true
		iris.StaticCacheDuration = time.Second
	}

What would be good design then?
